src/ORM/DataObject.php

```diff
@@ -4209,6 +4210,35 @@ public function mergeRelatedObjects($list, $items)
         return $added;
     }
 
+    /**
+     * Generate a unique key for data object
+     * the unique key uses the @see DataObject::getCacheKeyComponent() extension point so unique key modifiers
+     * such as versioned or fluent are covered
+     * i.e. same data object in different stages or different locales will produce different unique key
+     *
+     * recommended use:
+     * - when you need unique key for caching purposes
+     * - when you need unique id on the front end (for example JavaScript needs to target specific element)
+     *
+     * @return string
+     */
+    public function getUniqueKey(): string
+    {
+        // extract class name (remove namespaces)
+        $classSegments = explode('\\', $this->ClassName);
+
+        if (count($classSegments) === 0) {
+            return '';
+        }
+
+        $class = array_pop($classSegments);
```

Comment:
What's the rational for not using the fully qualified class name here?

Reply:
actually, the fully qualified class name is used in generating the hash, the class name and id are mostly for readability purposes. The key is already unique without the class name and id.